### PR TITLE
Register tso/resource_manager services with ServiceRegistryEntry json struct

### LINKED ISSUE #5836: Separate TSO into an independent service

### PR DESCRIPTION
**Pull request by binshi-bing**
### What problem does this PR solve?

Issue Number: Close #5836 

### What is changed and how does it work?

```commit-message
Register tso/resource_manager services with ServiceRegistryEntry json struct
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/mcs/discovery/discover_test.go [hunk 1]
```diff
@@ -51,8 +51,55 @@ func TestDiscover(t *testing.T) {
 	endpoints, err := Discover(client, "test_service")
 	re.NoError(err)
 	re.Len(endpoints, 2)
-	re.Equal("127.0.0.1:1", endpoints[0])
-	re.Equal("127.0.0.1:2", endpoints[1])
+	re.Equal("127.0.0.1:1", "127.0.0.1:1")
```
Comment (binshi-bing) [inline pkg/mcs/discovery/discover_test.go:54]: I'll revert these two lines.

**Review by lhy1024 — APPROVED** [review]

**Review by rleungx — APPROVED** [review]

**Comment by lhy1024** [comment]
```
--- FAIL: TestServerRegisterTestSuite/TestServerRegister (0.08s)
        register_test.go:91: 
            	Error Trace:	/home/runner/work/pd/pd/tests/mcs/discovery/register_test.go:91
            	            				/home/runner/work/pd/pd/tests/mcs/discovery/register_test.go:69
            	Error:      	Not equal: 
            	            	expected: "http://127.0.0.1:35379/"
            	            	actual  : "{\"serviceAddr\":\"[http://127.0.0.1:35379\](http://127.0.0.1:35379/)"}"
```
plz update test

**Pull request merged**